### PR TITLE
fix(button-base): make aria label change on update after first update lifecycle

### PR DESCRIPTION
**Pull request by SavkaTaras**
## Description
Currently the aria-label attribute gets set on firstUpdate. This causes an issue when the button is rendered and then the prop is updated. The change is not getting reflected. This makes sure the change actually gets reflected. 

GitHub bug:

## Motivation and context
Adobe Expres bug

## Related issue(s)

-   fixes https://github.com/adobe/spectrum-web-components/issues/5620

## Screenshots (if appropriate)

---

## Author's checklist

-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** and **[PULL_REQUESTS](<(https://github.com/adobe/spectrum-web-components/blob/main/PULL_REQUESTS.md)>)** documents.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)
-   [ ] I have added automated tests to cover my changes.
-   [x] I have included a well-written changeset if my change needs to be published.
-   [ ] I have included updated documentation if my change required it.

---

## Reviewer's checklist

-   [x] Includes a Github Issue with appropriate flag or Jira ticket number without a link
-   [ ] Includes thoughtfully written changeset if changes suggested include `patch`, `minor`, or `major` features
-   [ ] Automated tests cover all use cases and follow best practices for writing
-   [ ] Validated on all supported browsers
-   [ ] All VRTs are approved before the author can update Golden Hash

### Manual review test cases

-   [ ] _Descriptive Test Statement_

    1. Go [here](url)
    2. Do this action
    3. Expect this result

-   [ ] _Descriptive Test Statement_
    1. Go [here](url)
    2. Do this action
    3. Expect this result

### Device review

-   [ ] Did it pass in Desktop?
-   [ ] Did it pass in (emulated) Mobile?
-   [ ] Did it pass in (emulated) iPad?

### CODE REVIEW

File: packages/button/src/ButtonBase.ts [hunk 1]
```diff
@@ -255,6 +248,14 @@ export class ButtonBase extends ObserveSlotText(LikeAnchor(Focusable), '', [
...
             this.anchorElement.addEventListener('focus', this.proxyFocus);
         }
+
+        if (changed.has('label')) {
+            if (this.label) {
+                this.setAttribute('aria-label', this.label);
+            } else {
+                this.removeAttribute('aria-label');
+            }
+        }
```
Comment (Rajdeepc) [inline packages/button/src/ButtonBase.ts:258]: Thanks for doing this. Its a miss edge case from our side. But for better code maintainability can you extract this logic into a private method to avoid duplication. There are three instances of the same code. Second can you please add a test to verify `aria-label` changes when the `label` property  changes after initial render.

**Review by blunteshwar — COMMENTED** [review]
LGTM! 
Once we extract the logic in a function I believe we are good to mege!

**Review by caseyisonit — CHANGES_REQUESTED** [review]
@tsavka-adobe Theres a few things that need to be addressed with this PR before we can get it merged in. 

1. [We will need you to sign the CLA ](https://github.com/adobe/spectrum-web-components/pull/5619/checks?check_run_id=47228796379)
2. There are failing tests that need to be resolved
3. Automated tests need to be added to cover this new functionality
4. Manual testing steps need to be added to the PR descriptions particularly since this is a11y based

**Comment by tsavka-adobe** [comment]
please close this PR. I think there is another PR for this issue.
